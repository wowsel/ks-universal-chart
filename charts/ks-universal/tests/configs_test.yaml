# tests/configs/configs_test.yaml
suite: test configurations
templates:
  - configs.yaml
tests:
  - it: should create basic config map and secret
    values:
      - ./configs/values/basic-config.yaml
    asserts:
      - matchRegex:
          path: kind
          pattern: ConfigMap
        documentIndex: 0
      - matchRegex:
          path: kind
          pattern: Secret
        documentIndex: 1
      - equal:
          path: metadata.name
          value: app-config
        documentIndex: 0
      - equal:
          path: metadata.name
          value: app-secrets
        documentIndex: 1
      - equal:
          path: data["app.env"]
          value: production
        documentIndex: 0
      - equal:
          path: data["app.debug"]
          value: "false"
        documentIndex: 0
      - matchRegex:
          path: data["session.key"]
          pattern: "^[A-Za-z0-9+/=]+$"
        documentIndex: 1
      - matchRegex:
          path: data["encryption.key"]
          pattern: "^[A-Za-z0-9+/=]+$"
        documentIndex: 1

  - it: should create database config and credentials
    values:
      - ./configs/values/db-config.yaml
    asserts:
      - matchRegex:
          path: kind
          pattern: ConfigMap
        documentIndex: 0
      - matchRegex:
          path: kind
          pattern: Secret
        documentIndex: 1
      - equal:
          path: metadata.name
          value: db-config
        documentIndex: 0
      - equal:
          path: metadata.name
          value: db-credentials
        documentIndex: 1
      - equal:
          path: data.host
          value: postgresql.default.svc.cluster.local
        documentIndex: 0
      - equal:
          path: data.port
          value: "5432"
        documentIndex: 0
      - matchRegex:
          path: data.password
          pattern: "^[A-Za-z0-9+/=]+$"
        documentIndex: 1

  - it: should create redis config with multiline content
    values:
      - ./configs/values/mixed-config.yaml
    asserts:
      - equal:
          path: kind
          value: ConfigMap
        documentIndex: 0
      - equal:
          path: metadata.name
          value: app-mixed-config
        documentIndex: 0
      - matchRegex:
          path: data["settings.json"]
          pattern: '"cache":\s*{[^}]*"enabled":\s*true'
        documentIndex: 0
      - equal:
          path: kind
          value: ConfigMap
        documentIndex: 1
      - equal:
          path: metadata.name
          value: redis-config
        documentIndex: 1
      - equal:
          path: data
          value:
            redis.conf: |
              maxmemory 256mb
              maxmemory-policy allkeys-lru
        documentIndex: 1