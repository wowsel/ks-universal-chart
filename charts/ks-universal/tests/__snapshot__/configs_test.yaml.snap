should create configmaps from configs:
  1: |
    app.properties: |
      property1=value1
      property2=value2
    config.yaml: |
      key1: value1
      key2: value2
